AddonTerraformTemplate/main.tf

```diff
@@ -0,0 +1,448 @@
+  location            = azurerm_resource_group.rg.location
+  name                = var.cluster_name
+  resource_group_name = azurerm_resource_group.rg.name
+
+
+  dns_prefix          = var.dns_prefix
+  tags                = {
+    Environment = "Development"
+  }
+
+  default_node_pool {
+    name       = "agentpool"
+    vm_size    = "Standard_D2_v2"
+    node_count = var.agent_count
+  }
+
+  monitor_metrics {
+    annotations_allowed = "string"
+    labels_allowed = "string"
+  }
+
+  linux_profile {
+    admin_username = "ubuntu"
+
+    ssh_key {
+      key_data = file(var.ssh_public_key)
+    }
+  }
+  network_profile {
+    network_plugin    = "kubenet"
+    load_balancer_sku = "standard"
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+resource "azurerm_monitor_data_collection_endpoint" "example" {
+  name                          = "example-mdce"
+  resource_group_name           = azurerm_resource_group.rg.name
+  location                      = azurerm_resource_group.rg.location
+  kind                          = "Linux"
+}
+resource "azurerm_monitor_data_collection_rule" "example" {
+  name                = "example-rule"
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = azurerm_resource_group.rg.location
+  data_collection_endpoint_id = azurerm_monitor_data_collection_endpoint.example.id
+  kind                          = "Linux"
+
+  destinations {
+    monitor_account {
+      monitor_account_id = var.monitor_workspace_id
+      name = "MonitoringAccount1"
+    }
+  }
+
+  data_flow {
+    streams      = ["Microsoft-PrometheusMetrics"]
+    destinations = ["MonitoringAccount1"]
+  }
+
+
+  data_sources {
+    prometheus_forwarder {
+      streams                       = ["Microsoft-PrometheusMetrics"]
+      name                          = "PrometheusDataSource"
+    }
+  }
+
+  description = "DCR for Azure Monitor Metrics Profile (Managed Prometheus)"
+  depends_on = [
+    azurerm_monitor_data_collection_endpoint.example
+  ]
+}
+resource "azurerm_monitor_data_collection_rule_association" "example2" {
+  target_resource_id          = azurerm_kubernetes_cluster.k8s.id
+  data_collection_endpoint_id = azurerm_monitor_data_collection_endpoint.example.id
+  description                 = "Association of data collection rule. Deleting this association will break the data collection for this AKS Cluster."
+}
+resource "azurerm_dashboard_grafana" "example" {
+  name                              = "example-dg"
+  resource_group_name               = azurerm_resource_group.rg.name
+  location                          = azurerm_resource_group.rg.location
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  azure_monitor_workspace_integrations {
+    resource_id  = var.monitor_workspace_id
+  }
+
+}
+
+resource "azapi_resource" "NodeRecordingRulesRuleGroup" {
+  type = "Microsoft.AlertsManagement/prometheusRuleGroups@2023-03-01"
+  name = "example-NodeRecordingRulesRuleGroup"
```

Comment:
Any reason as to why the name's starting with 'example-'? We should keep the default names consistant with the other forms of onboarding (ARM, Bicep, Policy, CLI, UX) i.e. 'NodeRecordingRulesRuleGroup' imo

Reply:
Thanks Kaveesh! Renaming this to 'NodeRecordingRulesRuleGroup' as discussed.